runsc/container/container_test.go

```diff
@@ -3127,3 +3129,160 @@ func TestOverlayByMountAnnotation(t *testing.T) {
+		}
+	}
+	bigFile.Close()
+
+	// Create a test script which can be used to get the checksums
+	// from a specified directory.
+	scriptFile := path.Join(sourceDir, "test-script")
+	f, err := os.OpenFile(scriptFile, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		t.Fatalf("os.OpenFile() failed: %v", err)
+	}
+	if _, err := f.Write([]byte(`#!/bin/bash
+set -u -e -o pipefail
+dir=$1
+find $dir -printf "%P\n" | sort | md5sum
+find $dir -type f | sort | xargs cat | md5sum
+find $dir -type l | sort | xargs readlink | md5sum`)); err != nil {
+		t.Fatalf("Write() failed: %v", err)
+	}
+	f.Close()
+
+	// Create the EROFS images in a temporary directory.
+	imageDir, err := ioutil.TempDir(testutil.TmpDir(), "erofs-images")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir() failed: %v", err)
+	}
+
+	images := []struct {
+		name string
+		cmd  string
+	}{
+		{
+			// Generate extended inodes. Inline regular files if possible.
+			name: "image1",
+			cmd:  "mkfs.erofs -E force-inode-extended",
+		},
+		{
+			// Generate extended inodes. Do not inline regular files.
+			name: "image2",
+			cmd:  "mkfs.erofs -E force-inode-extended -E noinline_data",
+		},
+		{
+			// Generate compact inodes. Inline regular files if possible.
+			name: "image3",
+			cmd:  "mkfs.erofs -E force-inode-compact",
+		},
+		{
+			// Generate compact inodes. Do not inline regular files.
+			name: "image4",
+			cmd:  "mkfs.erofs -E force-inode-compact -E noinline_data",
+		},
+	}
+
+	for _, i := range images {
+		cmd := fmt.Sprintf("%s %s %s", i.cmd, path.Join(imageDir, i.name), sourceDir)
+		if out, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput(); err != nil {
+			t.Fatalf("exec: sh -c %q, err: %v, out: %s", cmd, err, out)
+		}
+	}
+
+	var checksums string
+
+	// Get the checksums from the source directory on host.
+	if out, err := exec.Command(scriptFile, sourceDir).CombinedOutput(); err != nil {
+		t.Fatalf("exec: %s %s, err: %v, out: %s", scriptFile, sourceDir, err, out)
+	} else {
+		checksums = string(out)
+	}
+
+	for name, conf := range configs(t, false /* noOverlay */) {
+		t.Run(name, func(t *testing.T) {
+			spec, _ := sleepSpecConf(t)
+			_, bundleDir, cleanup, err := testutil.SetupContainer(spec, conf)
+			if err != nil {
+				t.Fatalf("error setting up container: %v", err)
+			}
+			defer cleanup()
+
+			// Create and start the container.
+			args := Args{
+				ID:        testutil.RandomContainerID(),
+				Spec:      spec,
+				BundleDir: bundleDir,
+			}
+			c, err := New(conf, args)
+			if err != nil {
+				t.Fatalf("error creating container: %v", err)
+			}
+			defer c.Destroy()
+			if err := c.Start(conf); err != nil {
+				t.Fatalf("error starting container: %v", err)
+			}
+
+			for _, i := range images {
+				// Create a temporary directory on host, and mount the EROFS image on it in container.
+				targetDir, err := ioutil.TempDir(testutil.TmpDir(), "erofs-target")
+				if err != nil {
+					t.Fatalf("ioutil.TempDir() failed: %v", err)
+				}
```

Comment:
Why do we need to create temporary directory on the host? IIUC, this is the "destination", which will be created inside the container. You can just use destination as `/erofs1`, `/erofs2`, ...

Reply:
Yeah, you are right. This is the "destination" and can be created inside the container. Will update it in the next version.